plugins/redis/redis.cpp

```diff
@@ -293,34 +388,40 @@ void * Connection::redisCommand(const char * format, ...)
     //Copied from https://github.com/redis/hiredis/blob/master/hiredis.c ~line:1008 void * redisCommand(redisContext * context, const char * format, ...)
     //with redisSetTimeout(); added.
     va_list parameters;
-    void * reply = NULL;
+    void * reply = nullptr;
     va_start(parameters, format);
-    redisSetTimeout();
+    assertTimeout(redisSetTimeout());
     reply = ::redisvCommand(context, format, parameters);
     va_end(parameters);
     return reply;
 }
-unsigned Connection::timeLeft()
+int Connection::setTimeout(int _timeout)
+{
+    struct timeval to = { _timeout/1000, (_timeout%1000)*1000 };
+    if (!context)
+        return REDIS_ERR;
+    return ::redisSetTimeout(context, to);//NOTE: ::redisSetTimeout sets the socket timeout and therefore 0 => forever
+}
+int Connection::redisSetTimeout()
 {
-    unsigned _timeLeft = timeout.timeLeft();
+    int _timeLeft = (int)timeLeft();
     if (_timeLeft == 0 && timeout.getTimeout() != 0)
-        ::rtlFail(0, "Redis Plugin: ERROR - function timed out internally.");
-    return _timeLeft;
+        return INTERNAL_TIMEOUT;
+    return setTimeout(_timeLeft);
 }
-void Connection::redisSetTimeout()
+int SubConnection::redisSetUnsubscribeTimeout()
 {
+    getUnsubscribeTimeout();
```

Comment:
minor:  should save the value in a local variable, and use that in the rest of the function, rather than the member variable.